pkg/engine/params.go

```diff
@@ -219,20 +219,20 @@ func getParameters(cs *api.ContainerService, generatorCode string, aksEngineVers
 	if properties.HasWindows() {
 		addValue(parametersMap, "windowsAdminUsername", properties.WindowsProfile.AdminUsername)
 		addSecret(parametersMap, "windowsAdminPassword", properties.WindowsProfile.AdminPassword, false)
-		if properties.WindowsProfile.ImageVersion != "" {
```

Comment:
The changes here are merely conditional logical optimizations to account for the new `HasImageRef()` condition, right? Pre-existing api model image configuration patterns should still work, correct?

Reply:
Correct. Existing API model patterns should work and also all default behaviors should remain the same.